PocketBeagle/TechLab/counter.js

```diff
@@ -1,56 +1,70 @@
 #!/usr/bin/env node
-var b = require('bonescript');
+// Pre-steps (may need to eventually put in this program):
+//   * Make sure /sys/class/gpio/gpio45 is exported
+//   * Make sure /sys/class/gpio/gpio117 is exported
+//   * Make sure pinmux is in gpio mode
+//   * Make sure gpio is in input mode
 
-var L_BUTTON = "P2_33";
-var R_BUTTON = "P1_29";
+var Epoll = require('epoll').Epoll;
+var fs = require('fs');
+
+var L_BUTTON_PATH = "/sys/class/gpio/gpio45"; // P2_33
+var R_BUTTON_PATH = "/sys/class/gpio/gpio117"; // P1_29
 var debounceActive = false;
 var counter = 0;
+var buffer = Buffer.alloc(1);
 
 writeDigit(0, getDigit(0));
 writeDigit(1, getDigit(0));
 
-b.pinMode(L_BUTTON, b.INPUT, 7, null, null);
-b.attachInterrupt(L_BUTTON, true, b.FALLING, onPress);
-b.pinMode(R_BUTTON, b.INPUT, 7, null, null);
-b.attachInterrupt(R_BUTTON, true, b.FALLING, onPress);
+var L_BUTTON = fs.openSync(L_BUTTON_PATH + "/value", "r");
+var R_BUTTON = fs.openSync(R_BUTTON_PATH + "/value", "r");
+
+// Set interrupt mode
+fs.writeFileSync(L_BUTTON_PATH + "/edge", "falling");
+fs.writeFileSync(R_BUTTON_PATH + "/edge", "falling");
+
+// Attach interrupts
+var left_poller = new Epoll(leftPress);
+left_poller.add(L_BUTTON, Epoll.EPOLLPRI);
+console.log("Interrupt handler for L attached");
+
+var right_poller = new Epoll(rightPress);
+right_poller.add(R_BUTTON, Epoll.EPOLLPRI);
+console.log("Interrupt handler for R attached");
+
 console.log('Hit ^C to exit');
 
-function onPress(err, x) {
```

Comment:
I see why you split into two functions (that's what the Epoll API wants).

What if you kept a single `onPress` function that accepts button as an arg, then used [`bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind)?

The advantage here is a reduction in duplicated code (`setTimeout(clearDebounce, 150)`, etc.).

The disadvantage is a slight drop in efficiency because `bind` introduces an intermediate scope. This drop is usually negligible, however.

Up to you what to do, but wanted to raise the possibility.

Reply:
That is a good suggestion, I hadn't even considered using something like the `bind` method. However, I am thinking that it may be better to keep the two separate functions. My reasoning is that this is one of the first examples a new user would see, and using the `bind` method may add a level of complexity (especially for someone without much Javascript experience). That being said, I'm interested in hearing your thoughts on this.

Reply:
My thoughts were that I wanted to hear your thoughts. I think that this argument (prefer legibility over efficiency in code for a beginner) is an excellent thought :)